### PR TITLE
Test `typed_model` utilities

### PR DESCRIPTION
**Pull request by strickvl**
## Describe changes
I implemented/fixed _ to achieve _.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by fa9r — APPROVED** [review]

File: tests/unit/utils/test_typed_model_utils.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    matrix = TheMatrix(choice=RedPill())
+    d = matrix.dict()
+    new_matrix = TheMatrix.parse_obj(d)
+    assert isinstance(new_matrix.choice, RedPill)
+    assert new_matrix.choice.type
+    assert (
+        new_matrix.choice.type
+        == "tests.unit.utils.test_typed_model_utils.test_basetypedmodel_works.<locals>.RedPill"
+    )
```
Comment (fa9r) [inline tests/unit/utils/test_typed_model_utils.py:42]: Actually, perhaps this is good enough?

```suggestion
    assert new_matrix.choice.type.endswith("RedPill")
```

**Comment by fa9r** [comment]
What does the BaseTypedModel do exactly? Still not sure I fully get it. Perhaps it could be useful for #1151

**Comment by strickvl** [comment]
@fa9r Not really sure. @stefannica added it for something he was writing.

**Review by stefannica — APPROVED** [review]

File: tests/unit/utils/test_typed_model_utils.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#  permissions and limitations under the License.
+
+from typing import Union
+
+from pydantic import Field
+
+from zenml.utils.typed_model import BaseTypedModel
+
+
+def test_basetypedmodel_works():
```
[the remark this answers is not in the document]
Comment (stefannica) [reply tests/unit/utils/test_typed_model_utils.py:22]: Scratch that, I just realized you're already doing it without saving the file on disk (which you don't need to do anyway).

**Pull request merged**